raiden/tests/integration/test_settlement.py

```diff
@@ -104,29 +106,44 @@ def test_settlement(raiden_network, settle_timeout, reveal_timeout):
 
     # a ChannelClose event will be generated, this will be polled by both apps
     # and each must start a task for calling settle
-    channel0.external_state.netting_channel.close(
-        app0.raiden.address,
+    channel1.external_state.netting_channel.close(
+        app1.raiden.address,
         transfermessage,
         None,
     )
+    wait_until_block(chain0, chain0.block_number() + 1)
+    assert channel0.external_state.closed_block != 0
+    assert channel1.external_state.closed_block != 0
+    assert channel0.external_state.settled_block == 0
+    assert channel1.external_state.settled_block == 0
 
     # unlock will not be called by Channel.channel_closed because we did not
     # register the secret
-    channel0.external_state.netting_channel.unlock(
-        app0.raiden.address,
+    assert lock.expiration > app0.raiden.chain.block_number()
+    assert lock.hashlock == sha3(secret)
+
+    channel1.external_state.netting_channel.unlock(
+        app1.raiden.address,
         [unlock_proof],
     )
 
-    settle_expiration = chain0.block_number() + settle_timeout
+    settle_expiration = chain0.block_number() + settle_timeout + 2
     wait_until_block(chain0, settle_expiration)
+    gevent.sleep(1)
```

Comment:
Why are we waiting for a second here? If it's for the event polling we can use the channel settle_event instead `channel0.settle_event.wait()`